File: Ordenacao/BubbleSort.cpp
#include <iostream>
#include <vector>

void BubbleSort(std::vector<int> &vect)
{
    int n = vect.size();
    bool swapped;
    for (int i = n - 1; i > 0; i--)
    {
        swapped = false;
        for (int j = 0; j < i; j++)
        {
            if (vect[j] > vect[j + 1])
            {
                int aux = vect[j + 1];
                vect[j + 1] = vect[j];
                vect[j] = aux;
                swapped = true;
            }
        }

        if (!swapped)
            return;
    }
}

int main()
{
    std::vector<int> vetor = {9, 3, 2, 4, 8, 1};
    BubbleSort(vetor);
    for (int x : vetor)
    {
        std::cout << x << " ";
    }

    return 0;
}

File: javaa/Ordenacao/BubbleSort.java
package javaa.Ordenacao;

public class BubbleSort {

  public static void sort(int[] array) {
    boolean swapped;

    for (int i = array.length - 1; i > 0; --i) {
      swapped = false;
      for (int j = 0; j < i; j++) {
        if (array[j] > array[j + 1]) {
          swapped = true;
          int aux = array[j + 1];
          array[j + 1] = array[j];
          array[j] = aux;
        }
      }
      if (!swapped) {
        break;
      }

    }

  }

  public static void main(String[] args) {

    int[] arr = { 1, 12, 22, 3, 4 };

    sort(arr);

    for (var i : arr) {
      System.out.println(i);
    }
  }
}
